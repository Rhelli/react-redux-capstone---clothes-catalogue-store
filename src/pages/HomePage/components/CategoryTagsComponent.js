/* eslint-disable jsx-a11y/no-noninteractive-element-interactions */
/* eslint-disable jsx-a11y/no-static-element-interactions */
/* eslint-disable jsx-a11y/click-events-have-key-events */
import React from 'react';
import PropTypes from 'prop-types';

const CategoryTagsComponent = props => {
  const {
    enabledTags, enableSelectedFilter,
  } = props;

  return (
    <div>
      {() => {
        if (enabledTags.genderFilter.male) {
          return (
            <div
              data-name="male"
              onClick={event => enableSelectedFilter(event, 'genderFilter')}
            >
              <h6 data-name="male">Mens</h6>
            </div>
          );
        }
        return (
          null
        );
      }}
      {() => {
        if (enabledTags.genderFilter.female) {
          return (
            <div
              data-name="female"
              onClick={event => enableSelectedFilter(event, 'genderFilter')}
            >
              <h6 data-name="female">Womens</h6>
            </div>
          );
        }
        return (
          null
        );
      }}
      {() => {
        if (enabledTags.clothesFilter.jackets) {
          return (
            <div
              data-name="jackets"
              onClick={event => enableSelectedFilter(event, 'clothesFilter')}
            >
              <h6 data-name="jackets">Coats & Jackets</h6>
            </div>
          );
        }
        return (
          null
        );
      }}
      {() => {
        if (enabledTags.clothesFilter.jeans) {
          return (
            <div
              data-name="jeans"
              onClick={event => enableSelectedFilter(event, 'clothesFilter')}
            >
              <h6 data-name="jeans">Jeans</h6>
            </div>
          );
        }
        return (
          null
        );
      }}
      {() => {
        if (enabledTags.clothesFilter.shirts) {
          return (
            <div
              data-name="shirts"
              onClick={event => enableSelectedFilter(event, 'clothesFilter')}
            >
              <h6 data-name="shirts">Shirts</h6>
            </div>
          );
        }
        return (
          null
        );
      }}
      {() => {
        if (enabledTags.clothesFilter.tops) {
          return (
            <div
              data-name="tops"
              onClick={event => enableSelectedFilter(event, 'clothesFilter')}
            >
              <h6 data-name="tops">Tops</h6>
            </div>
          );
        }
        return (
          null
        );
      }}
      {() => {
        if (enabledTags.clothesFilter.dresses) {
          return (
            <div
              data-name="dresses"
              onClick={event => enableSelectedFilter(event, 'clothesFilter')}
            >
              <h6 data-name="dresses">Dresses</h6>
            </div>
          );
        }
        return (
          null
        );
      }}
      {() => {
        if (enabledTags.colorFilter.black) {
          return (
            <div
              data-name="black"
              onClick={event => enableSelectedFilter(event, 'colorFilter')}
            >
              <h6 data-name="black">Black</h6>
            </div>
          );
        }
        return (
          null
        );
      }}
      {() => {
        if (enabledTags.colorFilter.darkestSpruce) {
          return (
            <div
              data-name="darkestSpruce"
              onClick={event => enableSelectedFilter(event, 'colorFilter')}
            >
              <h6 data-name="darkestSpruce">Darkest Spruce</h6>
            </div>
          );
        }
        return (
          null
        );
      }}
      {() => {
        if (enabledTags.colorFilter.multi) {
          return (
            <div
              data-name="multi"
              onClick={event => enableSelectedFilter(event, 'colorFilter')}
            >
              <h6 data-name="multi">Multi</h6>
            </div>
          );
        }
        return (
          null
        );
      }}
      {() => {
        if (enabledTags.colorFilter.blue) {
          return (
            <div
              data-name="blue"
              onClick={event => enableSelectedFilter(event, 'colorFilter')}
            >
              <h6 data-name="blue">Blue</h6>
            </div>
          );
        }
        return (
          null
        );
      }}
      {() => {
        if (enabledTags.colorFilter.blueDenim) {
          return (
            <div
              data-name="blueDenim"
              onClick={event => enableSelectedFilter(event, 'colorFilter')}
            >
              <h6 data-name="blueDenim">Blue Denim</h6>
            </div>
          );
        }
        return (
          null
        );
      }}
      {() => {
        if (enabledTags.colorFilter.navy) {
          return (
            <div
              data-name="navy"
              onClick={event => enableSelectedFilter(event, 'colorFilter')}
            >
              <h6 data-name="navy">Navy</h6>
            </div>
          );
        }
        return (
          null
        );
      }}
      {() => {
        if (enabledTags.colorFilter.green) {
          return (
            <div
              data-name="green"
              onClick={event => enableSelectedFilter(event, 'colorFilter')}
            >
              <h6 data-name="green">Green</h6>
            </div>
          );
        }
        return (
          null
        );
      }}
      {() => {
        if (enabledTags.colorFilter.white) {
          return (
            <div
              data-name="white"
              onClick={event => enableSelectedFilter(event, 'colorFilter')}
            >
              <h6 data-name="white">White</h6>
            </div>
          );
        }
        return (
          null
        );
      }}
      {() => {
        if (enabledTags.colorFilter.grape) {
          return (
            <div
              data-name="grape"
              onClick={event => enableSelectedFilter(event, 'colorFilter')}
            >
              <h6 data-name="grape">Grape</h6>
            </div>
          );
        }
        return (
          null
        );
      }}
      {() => {
        if (enabledTags.colorFilter.khaki) {
          return (
            <div
              data-name="khaki"
              onClick={event => enableSelectedFilter(event, 'colorFilter')}
            >
              <h6 data-name="khaki">Khaki</h6>
            </div>
          );
        }
        return (
          null
        );
      }}
      {() => {
        if (enabledTags.colorFilter.grey) {
          return (
            <div
              data-name="grey"
              onClick={event => enableSelectedFilter(event, 'colorFilter')}
            >
              <h6 data-name="grey">Grey</h6>
            </div>
          );
        }
        return (
          null
        );
      }}
      {() => {
        if (enabledTags.colorFilter.vintageIndigo) {
          return (
            <div
              data-name="vintageIndigo"
              onClick={event => enableSelectedFilter(event, 'colorFilter')}
            >
              <h6 data-name="">Vintage Indigo</h6>
            </div>
          );
        }
        return (
          null
        );
      }}
      {() => {
        if (enabledTags.colorFilter.mineralBlue) {
          return (
            <div
              data-name="mineralBlue"
              onClick={event => enableSelectedFilter(event, 'colorFilter')}
            >
              <h6 data-name="mineralBlue">Mineral Blue</h6>
            </div>
          );
        }
        return (
          null
        );
      }}
      {() => {
        if (enabledTags.colorFilter.charcoal) {
          return (
            <div
              data-name="charcoal"
              onClick={event => enableSelectedFilter(event, 'colorFilter')}
            >
              <h6 data-name="charcoal">Charcoal</h6>
            </div>
          );
        }
        return (
          null
        );
      }}
      {() => {
        if (enabledTags.colorFilter.tan) {
          return (
            <div
              data-name="tan"
              onClick={event => enableSelectedFilter(event, 'colorFilter')}
            >
              <h6 data-name="tan">Tan</h6>
            </div>
          );
        }
        return (
          null
        );
      }}
      {() => {
        if (enabledTags.colorFilter.dustyBlue) {
          return (
            <div
              data-name="dustyBlue"
              onClick={event => enableSelectedFilter(event, 'colorFilter')}
            >
              <h6 data-name="dustyBlue">Dusty Blue</h6>
            </div>
          );
        }
        return (
          null
        );
      }}
      {() => {
        if (enabledTags.colorFilter.midwash) {
          return (
            <div
              data-name="midwash"
              onClick={event => enableSelectedFilter(event, 'colorFilter')}
            >
              <h6 data-name="midwash">Midwash</h6>
            </div>
          );
        }
        return (
          null
        );
      }}
      {() => {
        if (enabledTags.colorFilter.red) {
          return (
            <div
              data-name="red"
              onClick={event => enableSelectedFilter(event, 'colorFilter')}
            >
              <h6 data-name="red">Red</h6>
            </div>
          );
        }
        return (
          null
        );
      }}
      {() => {
        if (enabledTags.colorFilter.rose) {
          return (
            <div
              data-name="rose"
              onClick={event => enableSelectedFilter(event, 'colorFilter')}
            >
              <h6 data-name="rose">Rose</h6>
            </div>
          );
        }
        return (
          null
        );
      }}
      {() => {
        if (enabledTags.colorFilter.lilac) {
          return (
            <div
              data-name="lilac"
              onClick={event => enableSelectedFilter(event, 'colorFilter')}
            >
              <h6 data-name="lilac">Lilac</h6>
            </div>
          );
        }
        return (
          null
        );
      }}
    </div>
  );
};

CategoryTagsComponent.propTypes = {
  enableSelectedFilter: PropTypes.func.isRequired,
  enabledTags: PropTypes.shape({
    search: PropTypes.shape({
      input: PropTypes.string,
    }),
    clothesFilter: PropTypes.shape({
      jackets: PropTypes.bool.isRequired,
      jeans: PropTypes.bool.isRequired,
      shirts: PropTypes.bool.isRequired,
      tops: PropTypes.bool.isRequired,
      dresses: PropTypes.bool.isRequired,
    }).isRequired,
    genderFilter: PropTypes.shape({
      male: PropTypes.bool.isRequired,
      female: PropTypes.bool.isRequired,
    }).isRequired,
    colorFilter: PropTypes.shape({
      black: PropTypes.bool.isRequired,
      darkestSpruce: PropTypes.bool.isRequired,
      multi: PropTypes.bool.isRequired,
      blue: PropTypes.bool.isRequired,
      blueDenim: PropTypes.bool.isRequired,
      navy: PropTypes.bool.isRequired,
      green: PropTypes.bool.isRequired,
      white: PropTypes.bool.isRequired,
      grape: PropTypes.bool.isRequired,
      khaki: PropTypes.bool.isRequired,
      grey: PropTypes.bool.isRequired,
      vintageIndigo: PropTypes.bool.isRequired,
      mineralBlue: PropTypes.bool.isRequired,
      charcoal: PropTypes.bool.isRequired,
      tan: PropTypes.bool.isRequired,
      dustyBlue: PropTypes.bool.isRequired,
      midwash: PropTypes.bool.isRequired,
      red: PropTypes.bool.isRequired,
      rose: PropTypes.bool.isRequired,
      lilac: PropTypes.bool.isRequired,
    }).isRequired,
  }).isRequired,
};

export default CategoryTagsComponent;
