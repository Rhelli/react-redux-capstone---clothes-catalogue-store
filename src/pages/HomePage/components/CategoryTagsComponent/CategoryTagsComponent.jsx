/* eslint-disable jsx-a11y/no-noninteractive-element-interactions */
/* eslint-disable jsx-a11y/no-static-element-interactions */
/* eslint-disable jsx-a11y/click-events-have-key-events */
import React from 'react';
import PropTypes from 'prop-types';
import getColorPreview from '../../../../utils/colorUtil';
import styles from './CategoryTagsComponent.module.scss';

const CategoryTagsComponent = props => {
  const {
    categoryData, enableSelectedFilter,
  } = props;

  return (
    <div className={styles.categoryTagsContainer}>
      { categoryData.genderFilter.male ? (
        <div
          data-name="male"
          className={styles.male}
          onClick={event => enableSelectedFilter(event, 'genderFilter')}
        >
          <h6 data-name="male">Mens&nbsp;&nbsp;&nbsp;×</h6>
        </div>
      ) : null}
      { categoryData.genderFilter.female ? (
        <div
          data-name="female"
          className={styles.female}
          onClick={event => enableSelectedFilter(event, 'genderFilter')}
        >
          <h6 data-name="female">Womens&nbsp;&nbsp;&nbsp;×</h6>
        </div>
      ) : null}
      { categoryData.clothesFilter.jacket ? (
        <div
          data-name="jacket"
          className={styles.jacket}
          onClick={event => enableSelectedFilter(event, 'clothesFilter')}
        >
          <h6 data-name="jacket">Coats & Jackets&nbsp;&nbsp;&nbsp;×</h6>
        </div>
      ) : null}
      { categoryData.clothesFilter.jeans ? (
        <div
          data-name="jeans"
          className={styles.jeans}
          onClick={event => enableSelectedFilter(event, 'clothesFilter')}
        >
          <h6 data-name="jeans">Jeans&nbsp;&nbsp;&nbsp;×</h6>
        </div>
      ) : null}
      { categoryData.clothesFilter.shirts ? (
        <div
          data-name="shirts"
          className={styles.shirts}
          onClick={event => enableSelectedFilter(event, 'clothesFilter')}
        >
          <h6 data-name="shirts">Shirts&nbsp;&nbsp;&nbsp;×</h6>
        </div>
      ) : null}
      { categoryData.clothesFilter.tops ? (
        <div
          data-name="tops"
          className={styles.tops}
          onClick={event => enableSelectedFilter(event, 'clothesFilter')}
        >
          <h6 data-name="tops">Tops&nbsp;&nbsp;&nbsp;×</h6>
        </div>
      ) : null}
      { categoryData.clothesFilter.dresses ? (
        <div
          data-name="dresses"
          className={styles.dresses}
          onClick={event => enableSelectedFilter(event, 'clothesFilter')}
        >
          <h6 data-name="dresses">Dresses&nbsp;&nbsp;&nbsp;×</h6>
        </div>
      ) : null}
      { categoryData.colorFilter.black ? (
        <div
          data-name="black"
          style={getColorPreview('black')}
          onClick={event => enableSelectedFilter(event, 'colorFilter')}
        >
          <h6 data-name="black">Black&nbsp;&nbsp;&nbsp;×</h6>
        </div>
      ) : null}
      { categoryData.colorFilter.darkestSpruce ? (
        <div
          data-name="darkestSpruce"
          style={getColorPreview('darkestSpruce')}
          onClick={event => enableSelectedFilter(event, 'colorFilter')}
        >
          <h6 data-name="darkestSpruce">Darkest Spruce&nbsp;&nbsp;&nbsp;×</h6>
        </div>
      ) : null}
      { categoryData.colorFilter.multi ? (
        <div
          data-name="multi"
          className={styles.multi}
          onClick={event => enableSelectedFilter(event, 'colorFilter')}
        >
          <h6 data-name="multi">Multi&nbsp;&nbsp;&nbsp;×</h6>
        </div>
      ) : null}
      { categoryData.colorFilter.blue ? (
        <div
          data-name="blue"
          style={getColorPreview('blue')}
          onClick={event => enableSelectedFilter(event, 'colorFilter')}
        >
          <h6 data-name="blue">Blue&nbsp;&nbsp;&nbsp;×</h6>
        </div>
      ) : null}
      { categoryData.colorFilter.blueDenim ? (
        <div
          data-name="blueDenim"
          style={getColorPreview('blueDenim')}
          onClick={event => enableSelectedFilter(event, 'colorFilter')}
        >
          <h6 data-name="blueDenim">Blue Denim&nbsp;&nbsp;&nbsp;×</h6>
        </div>
      ) : null }
      { categoryData.colorFilter.navy ? (
        <div
          data-name="navy"
          style={getColorPreview('navy')}
          onClick={event => enableSelectedFilter(event, 'colorFilter')}
        >
          <h6 data-name="navy">Navy&nbsp;&nbsp;&nbsp;×</h6>
        </div>
      ) : null}
      { categoryData.colorFilter.green ? (
        <div
          data-name="green"
          style={getColorPreview('green')}
          onClick={event => enableSelectedFilter(event, 'colorFilter')}
        >
          <h6 data-name="green">Green&nbsp;&nbsp;&nbsp;×</h6>
        </div>
      ) : null}
      { categoryData.colorFilter.white ? (
        <div
          data-name="white"
          style={getColorPreview('white')}
          onClick={event => enableSelectedFilter(event, 'colorFilter')}
        >
          <h6 data-name="white">White&nbsp;&nbsp;&nbsp;×</h6>
        </div>
      ) : null}
      { categoryData.colorFilter.grape ? (
        <div
          data-name="grape"
          style={getColorPreview('grape')}
          onClick={event => enableSelectedFilter(event, 'colorFilter')}
        >
          <h6 data-name="grape">Grape&nbsp;&nbsp;&nbsp;×</h6>
        </div>
      ) : null}
      { categoryData.colorFilter.khaki ? (
        <div
          data-name="khaki"
          style={getColorPreview('khaki')}
          onClick={event => enableSelectedFilter(event, 'colorFilter')}
        >
          <h6 data-name="khaki">Khaki&nbsp;&nbsp;&nbsp;×</h6>
        </div>
      ) : null}
      { categoryData.colorFilter.grey ? (
        <div
          data-name="grey"
          style={getColorPreview('grey')}
          onClick={event => enableSelectedFilter(event, 'colorFilter')}
        >
          <h6 data-name="grey">Grey&nbsp;&nbsp;&nbsp;×</h6>
        </div>
      ) : null}
      { categoryData.colorFilter.vintageIndigo ? (
        <div
          data-name="vintageIndigo"
          style={getColorPreview('vintageIndigo')}
          onClick={event => enableSelectedFilter(event, 'colorFilter')}
        >
          <h6 data-name="vintageIndigo">Vintage Indigo&nbsp;&nbsp;&nbsp;×</h6>
        </div>
      ) : null}
      { categoryData.colorFilter.mineralBlue ? (
        <div
          data-name="mineralBlue"
          style={getColorPreview('mineralBlue')}
          onClick={event => enableSelectedFilter(event, 'colorFilter')}
        >
          <h6 data-name="mineralBlue">Mineral Blue&nbsp;&nbsp;&nbsp;×</h6>
        </div>
      ) : null}
      { categoryData.colorFilter.charcoal ? (
        <div
          data-name="charcoal"
          style={getColorPreview('charcoal')}
          onClick={event => enableSelectedFilter(event, 'colorFilter')}
        >
          <h6 data-name="charcoal">Charcoal&nbsp;&nbsp;&nbsp;×</h6>
        </div>
      ) : null}
      { categoryData.colorFilter.tan ? (
        <div
          data-name="tan"
          style={getColorPreview('tan')}
          onClick={event => enableSelectedFilter(event, 'colorFilter')}
        >
          <h6 data-name="tan">Tan&nbsp;&nbsp;&nbsp;×</h6>
        </div>
      ) : null}
      { categoryData.colorFilter.dustyBlue ? (
        <div
          data-name="dustyBlue"
          style={getColorPreview('dustyBlue')}
          onClick={event => enableSelectedFilter(event, 'colorFilter')}
        >
          <h6 data-name="dustyBlue">Dusty Blue&nbsp;&nbsp;&nbsp;×</h6>
        </div>
      ) : null}
      { categoryData.colorFilter.midwash ? (
        <div
          data-name="midwash"
          style={getColorPreview('midwash')}
          onClick={event => enableSelectedFilter(event, 'colorFilter')}
        >
          <h6 data-name="midwash">Midwash&nbsp;&nbsp;&nbsp;×</h6>
        </div>
      ) : null}
      { categoryData.colorFilter.red ? (
        <div
          data-name="red"
          style={getColorPreview('red')}
          onClick={event => enableSelectedFilter(event, 'colorFilter')}
        >
          <h6 data-name="red">Red&nbsp;&nbsp;&nbsp;×</h6>
        </div>
      ) : null}
      { categoryData.colorFilter.rose ? (
        <div
          data-name="rose"
          style={getColorPreview('rose')}
          onClick={event => enableSelectedFilter(event, 'colorFilter')}
        >
          <h6 data-name="rose">Rose&nbsp;&nbsp;&nbsp;×</h6>
        </div>
      ) : null}
      { categoryData.colorFilter.lilac ? (
        <div
          data-name="lilac"
          style={getColorPreview('lilac')}
          onClick={event => enableSelectedFilter(event, 'colorFilter')}
        >
          <h6 data-name="lilac">Lilac&nbsp;&nbsp;&nbsp;×</h6>
        </div>
      ) : null}
    </div>
  );
};

CategoryTagsComponent.propTypes = {
  categoryData: PropTypes.shape({
    clothesFilter: PropTypes.shape({
      jacket: PropTypes.bool.isRequired,
      jeans: PropTypes.bool.isRequired,
      shirts: PropTypes.bool.isRequired,
      tops: PropTypes.bool.isRequired,
      dresses: PropTypes.bool.isRequired,
    }).isRequired,
    genderFilter: PropTypes.shape({
      male: PropTypes.bool.isRequired,
      female: PropTypes.bool.isRequired,
    }).isRequired,
    colorFilter: PropTypes.shape({
      black: PropTypes.bool.isRequired,
      darkestSpruce: PropTypes.bool.isRequired,
      multi: PropTypes.bool.isRequired,
      blue: PropTypes.bool.isRequired,
      blueDenim: PropTypes.bool.isRequired,
      navy: PropTypes.bool.isRequired,
      green: PropTypes.bool.isRequired,
      white: PropTypes.bool.isRequired,
      grape: PropTypes.bool.isRequired,
      khaki: PropTypes.bool.isRequired,
      grey: PropTypes.bool.isRequired,
      vintageIndigo: PropTypes.bool.isRequired,
      mineralBlue: PropTypes.bool.isRequired,
      charcoal: PropTypes.bool.isRequired,
      tan: PropTypes.bool.isRequired,
      dustyBlue: PropTypes.bool.isRequired,
      midwash: PropTypes.bool.isRequired,
      red: PropTypes.bool.isRequired,
      rose: PropTypes.bool.isRequired,
      lilac: PropTypes.bool.isRequired,
    }).isRequired,
  }).isRequired,
  enableSelectedFilter: PropTypes.func.isRequired,
};

export default CategoryTagsComponent;
