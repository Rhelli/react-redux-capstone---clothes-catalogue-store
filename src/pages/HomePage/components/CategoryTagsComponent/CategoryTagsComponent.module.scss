@-webkit-keyframes fade-in {
  0% {
    opacity: 0;
  }

  100% {
    opacity: 1;
  }
}

@keyframes fade-in {
  0% {
    opacity: 0;
  }

  100% {
    opacity: 1;
  }
}

.categoryTagsContainer {
  display: flex;
  flex-flow: row wrap;
  height: auto;
  padding: 8px 8px;
  background-color: #ddd4cc;
  border-top: 2px solid #beb5af;

  div {
    padding: 6px 8px;
    border-radius: 2px;
    margin: 0 8px 8px 0;
    -webkit-animation: fade-in 0.2s cubic-bezier(0.455, 0.03, 0.515, 0.955) both;
    animation: fade-in 0.2s cubic-bezier(0.455, 0.03, 0.515, 0.955) both;

    h6 {
      font: 700 10px/5px 'Source Sans Pro', monospace;
    }
  }

  .male {
    background-color: #5dade2;
    border-bottom: 2px solid #4682a9;
    color: #d6eaf8;
  }

  .female {
    background-color: #f8bbd0;
    border-bottom: 2px solid #ba8c9c;
    color: #5d464e;
  }

  .jacket {
    background-color: #f7dc6f;
    border-bottom: 2px solid #b9a553;
    color: #7b6e37;
  }

  .jeans {
    background-color: #e59866;
    border-bottom: 2px solid #ac724d;
    color: #563926;
  }

  .shirts {
    background-color: #cddc39;
    border-bottom: 2px solid #9aa52b;
    color: #808924;
  }

  .tops {
    background-color: #880e4f;
    border-bottom: 2px solid #660a3b;
    color: #e1c3d3;
  }

  .dresses {
    background-color: #4db6ac;
    border-bottom: 2px solid #3a8981;
    color: #d3edea;
  }

  .multi {
    background-image:
      linear-gradient(
        to right,
        #64b5f6,
        #4dd0e1 20%,
        #81c784 40%,
        #dce775 60%,
        #ffb74d 80%,
        #ba68c8
      );
    border-top: 2px solid #e1c3d3;
    border-right: 2px solid #ba68c8;
    border-bottom: 2px solid #81c784;
    border-left: 2px solid #64b5f6;
    border-radius: 2px;
  }
}

@media only screen and (min-width: 768px) {
  .categoryTagsContainer {
    div {
      padding: 8px 10px;

      h6 {
        font-size: 14px;
        line-height: 14px;
      }
    }
  }
}
