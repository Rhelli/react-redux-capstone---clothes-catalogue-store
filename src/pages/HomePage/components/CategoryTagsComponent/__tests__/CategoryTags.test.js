import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer';
import CategoryTagsComponent from '../CategoryTagsComponent';

const categoryData = {
  clothesFilter: {
    jacket: true,
    jeans: false,
    shirts: true,
    tops: false,
    dresses: true,
  },
  genderFilter: {
    male: false,
    female: true,
  },
  colorFilter: {
    black: false,
    darkestSpruce: true,
    multi: false,
    blue: true,
    blueDenim: false,
    navy: true,
    green: false,
    white: true,
    grape: false,
    khaki: true,
    grey: false,
    vintageIndigo: true,
    mineralBlue: false,
    charcoal: true,
    tan: false,
    dustyBlue: true,
    midwash: false,
    red: true,
    rose: false,
    lilac: true,
  },
};

const enableSelectedFilter = jest.fn();

const categoryTagsComponent = <CategoryTagsComponent categoryData={categoryData} enableSelectedFilter={enableSelectedFilter} />

describe('The CategoryTagsComponent', () => {
  it('Matches the CategoryTagsComponent snapshot', () => {
    const tree = renderer.create(categoryTagsComponent).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Is comprised of one master container component with the correct className and tag attributes', () => {
    const { container } = render(categoryTagsComponent);
    expect(container.firstChild.nodeName).toBe('DIV');
    expect(container.firstChild.classList.contains('categoryTagsContainer')).toBe(true);
  });

  it('Should contain 14 active tags with the provided categoryData preset above', () => {
    const { container } = render(categoryTagsComponent);
    expect(container.firstChild.childElementCount).toEqual(14);
  });

  it('Should render tags with the correct html attributes', () => {
    const { container } = render(categoryTagsComponent);
    expect(container.firstChild.firstChild).toHaveAttribute('data-name', 'female');
    expect(container.firstChild.firstChild.classList.contains('female')).toBe(true);
  });
});
