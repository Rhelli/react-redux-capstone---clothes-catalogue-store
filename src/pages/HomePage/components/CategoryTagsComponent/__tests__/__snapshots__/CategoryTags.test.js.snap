// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The CategoryTagsComponent Matches the CategoryTagsComponent snapshot 1`] = `
<div
  className="categoryTagsContainer"
>
  <div
    className="female"
    data-name="female"
    onClick={[Function]}
    onKeyPress={[Function]}
    role="button"
    tabIndex="-1"
  >
    <h6
      data-name="female"
    >
      Womens   ×
    </h6>
  </div>
  <div
    className="jacket"
    data-name="jacket"
    onClick={[Function]}
    onKeyPress={[Function]}
    role="button"
    tabIndex="-1"
  >
    <h6
      data-name="jacket"
    >
      Coats & Jackets   ×
    </h6>
  </div>
  <div
    className="shirts"
    data-name="shirts"
    onClick={[Function]}
    onKeyPress={[Function]}
    role="button"
    tabIndex="-1"
  >
    <h6
      data-name="shirts"
    >
      Shirts   ×
    </h6>
  </div>
  <div
    className="dresses"
    data-name="dresses"
    onClick={[Function]}
    onKeyPress={[Function]}
    role="button"
    tabIndex="-1"
  >
    <h6
      data-name="dresses"
    >
      Dresses   ×
    </h6>
  </div>
  <div
    data-name="darkestSpruce"
    onClick={[Function]}
    onKeyPress={[Function]}
    role="button"
    style={
      Object {
        "backgroundColor": "#0E6251",
        "borderBottom": "2px solid #0C5647",
        "color": "#A5C4BE",
      }
    }
    tabIndex="-1"
  >
    <h6
      data-name="darkestSpruce"
    >
      Darkest Spruce   ×
    </h6>
  </div>
  <div
    data-name="blue"
    onClick={[Function]}
    onKeyPress={[Function]}
    role="button"
    style={
      Object {
        "backgroundColor": "#3498DB",
        "borderBottom": "2px solid #2E85C0",
        "color": "#CCE5F6",
      }
    }
    tabIndex="-1"
  >
    <h6
      data-name="blue"
    >
      Blue   ×
    </h6>
  </div>
  <div
    data-name="navy"
    onClick={[Function]}
    onKeyPress={[Function]}
    role="button"
    style={
      Object {
        "backgroundColor": "#2C3E50",
        "borderBottom": "2px solid #212F3C",
        "color": "#CACFD3",
      }
    }
    tabIndex="-1"
  >
    <h6
      data-name="navy"
    >
      Navy   ×
    </h6>
  </div>
  <div
    data-name="white"
    onClick={[Function]}
    onKeyPress={[Function]}
    role="button"
    style={
      Object {
        "backgroundColor": "#ffffff",
        "borderBottom": "2px solid #DFDFDF",
        "color": "#303030",
      }
    }
    tabIndex="-1"
  >
    <h6
      data-name="white"
    >
      White   ×
    </h6>
  </div>
  <div
    data-name="khaki"
    onClick={[Function]}
    onKeyPress={[Function]}
    role="button"
    style={
      Object {
        "backgroundColor": "#AED581",
        "borderBottom": "2px solid #83A061",
        "color": "#6D8551",
      }
    }
    tabIndex="-1"
  >
    <h6
      data-name="khaki"
    >
      Khaki   ×
    </h6>
  </div>
  <div
    data-name="vintageIndigo"
    onClick={[Function]}
    onKeyPress={[Function]}
    role="button"
    style={
      Object {
        "backgroundColor": "#7986CB",
        "borderBottom": "2px solid #5B6598",
        "color": "#DDE1F2",
      }
    }
    tabIndex="-1"
  >
    <h6
      data-name="vintageIndigo"
    >
      Vintage Indigo   ×
    </h6>
  </div>
  <div
    data-name="charcoal"
    onClick={[Function]}
    onKeyPress={[Function]}
    role="button"
    style={
      Object {
        "backgroundColor": "#424242",
        "borderBottom": "2px solid #292929",
        "color": "#D0D0D0",
      }
    }
    tabIndex="-1"
  >
    <h6
      data-name="charcoal"
    >
      Charcoal   ×
    </h6>
  </div>
  <div
    data-name="dustyBlue"
    onClick={[Function]}
    onKeyPress={[Function]}
    role="button"
    style={
      Object {
        "backgroundColor": "#AED6F1",
        "borderBottom": "2px solid #83A1B5",
        "color": "#6D8697",
      }
    }
    tabIndex="-1"
  >
    <h6
      data-name="dustyBlue"
    >
      Dusty Blue   ×
    </h6>
  </div>
  <div
    data-name="red"
    onClick={[Function]}
    onKeyPress={[Function]}
    role="button"
    style={
      Object {
        "backgroundColor": "#E53935",
        "borderBottom": "2px solid #AC2B28",
        "color": "#F8CECC",
      }
    }
    tabIndex="-1"
  >
    <h6
      data-name="red"
    >
      Red   ×
    </h6>
  </div>
  <div
    data-name="lilac"
    onClick={[Function]}
    onKeyPress={[Function]}
    role="button"
    style={
      Object {
        "backgroundColor": "#E1BEE7",
        "borderBottom": "2px solid #A98FAD",
        "color": "#F7EFF9",
      }
    }
    tabIndex="-1"
  >
    <h6
      data-name="lilac"
    >
      Lilac   ×
    </h6>
  </div>
</div>
`;
